src/main/java/org/gridsuite/directory/server/DirectoryController.java

```diff
@@ -38,103 +39,92 @@ public DirectoryController(DirectoryService service) {
-        return ResponseEntity.ok().body(service.renameElement(elementUuid, newElementName, userId));
+    @GetMapping(value = "/elements")
+    @Operation(summary = "Get elements infos from ids given as parameters")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The elements information")})
+    public ResponseEntity<Flux<ElementAttributes>> getElements(@RequestParam("id") List<UUID> ids) {
+        return ResponseEntity.ok().body(service.getElements(ids));
     }
 
-    @PutMapping(value = "/directories/{elementUuid}/rights", consumes = MediaType.APPLICATION_JSON_VALUE)
-    @Operation(summary = "Modify element/directory's access rights")
+    @PutMapping(value = "/elements/{elementUuid}", consumes = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(summary = "Update element/directory")
     @ApiResponses(value = {
         @ApiResponse(responseCode = "200", description = "Element/directory was successfully updated"),
         @ApiResponse(responseCode = "404", description = "The element was not found"),
-        @ApiResponse(responseCode = "403", description = "Not authorized to update this element access rights")
+        @ApiResponse(responseCode = "403", description = "Not authorized to update this element")
     })
-    public ResponseEntity<Mono<Void>> setAccessRights(@PathVariable("elementUuid") UUID elementUuid,
-                                                      @RequestBody boolean isPrivate,
-                                                      @RequestHeader("userId") String userId) {
-        return ResponseEntity.ok().body(service.setAccessRights(elementUuid, isPrivate, userId));
+    public ResponseEntity<Mono<Void>> updateElement(@PathVariable("elementUuid") UUID elementUuid,
+                                                    @RequestBody ElementAttributes elementAttributes,
+                                                    @RequestHeader("userId") String userId) {
+        return ResponseEntity.ok().body(service.updateElement(elementUuid, elementAttributes, userId));
     }
 
-    @DeleteMapping(value = "/directories/{elementUuid}")
-    @Operation(summary = "Remove directory/element")
-    @ApiResponses(@ApiResponse(responseCode = "200", description = "Directory/element was successfully removed"))
-    public ResponseEntity<Mono<Void>> deleteElement(@PathVariable("elementUuid") UUID elementUuid,
-                                                    @RequestHeader("userId") String userId) {
-        return ResponseEntity.ok().body(service.deleteElement(elementUuid, userId));
+    @PostMapping(value = "/elements/{elementUuid}/notification")
+    @Operation(summary = "Create change element notification")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "The notification has been sent"),
+        @ApiResponse(responseCode = "400", description = "The notification type is unknown")
```

Comment:
Add @ApiResponse(responseCode = "404", description = "Element not found")

Reply:
Done